### DISCUSSION TITLE
How to compare the text of the email with the template?

### DISCUSSION THREAD

**Discussion by Grinnbob** [explosion/spaCy#6332] [Help: Best practices]
There is a task of comparing the text of the email with the template - the email can be written arbitrarily, but it must necessarily contain blocks from the template, for example:

1. name at the beginning of the letter
2. mention of the project
3. mention of the project niche
4. mention of cases for this niche
5. call to action
6. appointment etc.

Not all blocks can be detected by regular expressions, in some cases it is necessary to select whole phrases. Tell me what spaCy methods are the best to solve the problem? Should it be a NER problem or something else?

**Comment by ahalterman** [comment]
It sounds like no matter what you'll need new labeled data to make this work. You can use NER models on longer spans of text, but models can struggle with it because it's not always clear exactly which token to begin with (especially for something vague like "mention of use cases for this niche.") You could try splitting it into sentences and classifying each sentence, or you could try an NER model. There's a discussion of a similar task here: https://support.prodi.gy/t/annotation-for-argument-mining/601/18
